bundles/org.openhab.binding.cbus/src/main/java/org/openhab/binding/cbus/handler/CBusCGateHandler.java

```diff
@@ -0,0 +1,427 @@
+        logger.debug("Initializing CGate Bridge handler. {} {}", getThing().getThingTypeUID(), getThing().getUID());
+        CBusCGateConfiguration configuration = getConfigAs(CBusCGateConfiguration.class);
+        logger.debug("Using configuration {}", configuration);
+        if ("127.0.0.1".equals(configuration.ipAddress) || "localhost".equals(configuration.ipAddress)) {
+            this.ipAddress = InetAddress.getLoopbackAddress();
+        } else {
+            try {
+                this.ipAddress = InetAddress.getByName(configuration.ipAddress);
+            } catch (UnknownHostException e1) {
+                updateStatus(ThingStatus.UNINITIALIZED, ThingStatusDetail.HANDLER_INITIALIZING_ERROR,
+                        "IP Address not resolvable");
+                return;
+            }
+        }
+        InetAddress address = this.ipAddress;
+        if (address != null) {
+            logger.debug("CGate IP         {}.", address.getHostAddress());
+        }
+
+        keepAliveFuture = scheduler.scheduleWithFixedDelay(this::keepAlive, 0, 100, TimeUnit.SECONDS);
+    }
+
+    private void keepAlive() {
+        CGateSession session = cGateSession;
+        if (session == null || !session.isConnected()) {
+            if (!getThing().getStatus().equals(ThingStatus.ONLINE)) {
+                connect();
+            } else {
+                updateStatus();
+            }
+        }
+    }
+
+    private void connect() {
+        CGateSession cGateSession = this.cGateSession;
+        if (cGateSession == null) {
+            cGateSession = CGateInterface.connect(this.ipAddress, 20023, 20024, 20025, new CBusThreadPool());
+            cGateSession.registerEventCallback(new EventMonitor());
+            cGateSession.registerStatusChangeCallback(new StatusChangeMonitor());
+            this.cGateSession = cGateSession;
+        }
+        if (cGateSession.isConnected()) {
+            logger.debug("CGate session reports online");
+            updateStatus(ThingStatus.ONLINE);
+        } else {
+            try {
+                cGateSession.connect();
+                updateStatus();
+            } catch (CGateConnectException e) {
+                updateStatus();
+                logger.debug("Failed to connect to CGate: {}", e.getMessage());
+                try {
+                    cGateSession.close();
+                } catch (CGateException e2) {
+                    // We dont really care if an exception is thrown when clossing the connection after a failure
+                    // connecting.
+                }
+            }
+        }
+    }
+
+    private void updateStatus() {
+        ThingStatus lastStatus = getThing().getStatus();
+        CGateSession cGateSession = this.cGateSession;
+        if (cGateSession == null) {
+            return;
+        }
+        if (cGateSession.isConnected()) {
+            updateStatus(ThingStatus.ONLINE);
+        } else {
+            if (!lastStatus.equals(ThingStatus.OFFLINE)) {
```

Comment:
`ThingStatus` is an enum so you could compare with `==` or `!=` which could make the code more readable.

Reply:
Changed here and in CbusLightHandler